Sources/ArcGISToolkit/Components/LocationButton/LocationButton.swift

```diff
@@ -0,0 +1,342 @@
+        locationDisplay.autoPanMode = autoPanMode
+    }
+    
+    /// This should be called when the button is pressed.
+    func buttonAction() {
+        switch Action(status: status, autoPanOptions: autoPanOptions) {
+        case .start:
+            // If the datasource is a system location datasource, then request authorization.
+            if locationDisplay.dataSource is SystemLocationDataSource,
+               CLLocationManager.shared.authorizationStatus == .notDetermined {
+                CLLocationManager.shared.requestWhenInUseAuthorization()
+            }
+            Task {
+                // Start the datasource, set initial auto pan mode.
+                do {
+                    locationDisplay.autoPanMode = initialAutoPanMode
+                    try await locationDisplay.dataSource.start()
+                } catch {
+                    print("Error starting location display: \(error)")
+                }
+            }
+        case .stop:
+            Task { await hideLocationDisplay() }
+        case .autoPanCycle:
+            // Need to use the select method here so that last selected mode
+            // gets set.
+            select(autoPanMode: nextAutoPanMode(current: autoPanMode, initial: initialAutoPanMode))
+        case .none:
+            return
+        }
+    }
+    
+    /// The next auto pan mode to be used when cycling through auto pan modes.
+    func nextAutoPanMode(current: LocationDisplay.AutoPanMode, initial: LocationDisplay.AutoPanMode) -> LocationDisplay.AutoPanMode {
+        guard let index = autoPanOptions.firstIndex(of: current) else { return initial }
+        let nextIndex = index.advanced(by: 1) == autoPanOptions.endIndex ? autoPanOptions.startIndex : index.advanced(by: 1)
+        return autoPanOptions[nextIndex]
+    }
+    
+    /// Hides the location display.
+    func hideLocationDisplay() async {
+        await locationDisplay.dataSource.stop()
+    }
+}
+
+extension LocationButton {
+    /// The type of actions that can take place when the button is pressed.
+    enum Action {
+        /// Start the location display.
+        case start
+        /// Stop the location display.
+        case stop
+        /// Set the next auto pan mode for cycling through.
+        case autoPanCycle
+        
+        /// The action that should occur for the specified state.
+        init?(
+            status: LocationDataSource.Status,
+            autoPanOptions: [LocationDisplay.AutoPanMode]
+        ) {
+            // Decide the button behavior based on the status.
+            switch status {
+            case .stopped, .failedToStart:
+                self = .start
+            case .started:
+                if autoPanOptions.count < 2 {
+                    // If there were no non-off options specified then set it to off
+                    // if the status is started.
+                    self = .stop
+                } else {
+                    self = .autoPanCycle
+                }
```

Comment:
You could use an if expression here.

```suggestion
                self = if autoPanOptions.count < 2 {
                    // Since there were no non-off options specified then set it
                    // to off since the status is started.
                    .stop
                } else {
                    .autoPanCycle
                }
```